library/PostgreSQL/Binary/Decoder.hs

```diff
@@ -152,6 +153,47 @@ uuid :: Decoder UUID
 uuid =
   UUID.fromWords <$> intOfSize 4 <*> intOfSize 4 <*> intOfSize 4 <*> intOfSize 4
 
+{-# INLINABLE ipv4 #-}
+ipv4 :: Decoder Data.IPv4
+ipv4 =
+  (,,,) <$> intOfSize 1 <*> intOfSize 1 <*> intOfSize 1 <*> intOfSize 1
+
+{-# INLINABLE ipv6 #-}
+ipv6 :: Decoder Data.IPv6
+ipv6 =
+  (,,,,,,,) <$> intOfSize 2 <*> intOfSize 2 <*> intOfSize 2 <*> intOfSize 2 <*> intOfSize 2 <*> intOfSize 2 <*> intOfSize 2 <*> intOfSize 2
+
+{-# INLINABLE inet #-}
+inet :: Decoder Data.Inet
+inet = do
+  af <- intOfSize 1
+  netmask <- intOfSize 1
+  isCidr <- intOfSize 1
+  ipSize <- intOfSize 1
+  if af == Data.afInet
+    then do
+      ip <- ipv4
+      return $ inetIPv4FromBytes af netmask isCidr ipSize ip
+    else do
+      ip <- ipv6
+      return $ inetIPv6FromBytes af netmask isCidr ipSize ip
+
+{-# INLINABLE inetIPv4FromBytes #-}
+inetIPv4FromBytes:: Word8 -> Word8 -> Word8 -> Int8 -> Data.IPv4 -> Data.Inet
```

Comment:
Better move this into the `where` clause of `inet` for two reasons:

1. This function only exists for `inet`
2. The API is always cleaner, when all the top-level functions operate on the same level of abstraction

Same applies to `inetIPv6FromBytes`.